marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/CallBatcherImpl.java

```diff
@@ -466,18 +466,18 @@ private void initialize() {
         }
         if (initialized.getAndSet(true)) return;
         
-        if(inputType == Void.class) {
+        threadPool = new CallingThreadPoolExecutor(this, getThreadCount());
+        jobStartTime = Calendar.getInstance();
+        started.set(true);
+        
+        if(callArgsGenerator != null) {
             for (int i=0;i<getThreadCount(); i++) {
                     CallArgs newInput = callArgsGenerator.apply(null);
                     if(newInput != null) {
                         submitCall((CallArgsImpl) newInput);
```

Comment:
Before submitCall() here and on line 711, should there be a test that throws an "unsupported implementation of call arguments" exception if CallArgs is not an instance of CallArgsImpl?